config/fs.go

```diff
@@ -187,6 +187,9 @@ func parseFSConfig(cfg *Config) {
 	if cfg.FuseMetricsEmitWaitDurationSec == 0 {
 		cfg.FuseMetricsEmitWaitDurationSec = defaultFuseMetricsEmitWaitDurationSec
 	}
+	if cfg.MaxConcurrency <= 0 {
```

Comment:
The process segfaults on startup if this value is negative, since we attempt to set a negative value to the weighted semaphore. I feel like it might be better to fix the problem for the user instead of an ungraceful crash, but if we think I should change `<=` to `==` to follow convention in the rest of the file, feel free to lmk.

Reply:
Ah, so then if one wants to disable it they'd have to do so explicitly. I'm not positive that they should ever want to, but I suppose it could help to cover the use case. I'd have to refactor the function a bit to make this work, but it shouldn't be anything too difficult probably.

Reply:
Can we also add a comment in the code to explain what are different meaning of the values eg. 0 means default, >0 means exact value, -1 means disabled.